src/sidebar/components/test/group-list-test.js

```diff
@@ -454,5 +480,174 @@ describe('groupList', function() {
+      filtered_groups: [OPEN_GROUP],
+      expectedShowFeatured: true,
+    },
+    {
+      description:
+        'If logged in and there are only groups with membership do not show `Featured Groups`.',
+      isLoggedIn: true,
+      filtered_groups: [PUBLIC_GROUP],
+      expectedShowFeatured: false,
+    },
+    {
+      description:
+        'If logged out and there is a restricted group scoped to the page show it in `Featured Groups`.',
+      isLoggedIn: false,
+      filtered_groups: [RESTRICTED_GROUP],
+      expectedShowFeatured: true,
+    },
+    {
+      description:
+        'If logged out and there is an open group show it in `Featured Groups`.',
+      isLoggedIn: false,
+      filtered_groups: [OPEN_GROUP],
+      expectedShowFeatured: true,
+    },
+  ].forEach(
+    ({ description, isLoggedIn, filtered_groups, expectedShowFeatured }) => {
+      it(description, function() {
+        fakeStore.profile = () => {
+          return { userid: isLoggedIn ? 'userid' : null };
+        };
+        groups = filtered_groups;
+
+        const element = createGroupList();
+        const showFeaturedGroups = element.ctrl.showFeaturedGroups();
+        const showCurrentlyViewing = element.ctrl.showCurrentlyViewing();
+
+        assert.isTrue(showFeaturedGroups === expectedShowFeatured);
+        assert.isTrue(showCurrentlyViewing === false);
+      });
+    }
+  );
+
+  [
+    {
+      description:
+        'If logged in and the user is a member of the unscoped group show it in `My Groups`',
+      isLoggedIn: true,
+      filtered_groups: [RESTRICTED_UNSCOPED_MEMBER_GROUP],
+      expectedShowMyGroups: true,
+    },
+    {
+      description:
+        'If logged in and the user is a member of the group show it in `My Groups`',
+      isLoggedIn: true,
+      filtered_groups: [PUBLIC_GROUP],
+      expectedShowMyGroups: true,
+    },
+  ].forEach(
+    ({ description, isLoggedIn, filtered_groups, expectedShowMyGroups }) => {
+      it(description, function() {
+        fakeStore.profile = () => {
+          return { userid: isLoggedIn ? 'userid' : null };
+        };
+        groups = filtered_groups;
+
+        const element = createGroupList();
+        const showMyGroups = element.ctrl.showMyGroups();
+        const showCurrentlyViewing = element.ctrl.showCurrentlyViewing();
+
+        assert.isTrue(showMyGroups === expectedShowMyGroups);
+        assert.isTrue(showCurrentlyViewing === false);
+      });
+    }
+  );
+
+  [
+    {
+      description:
+        'If logged out and there is a restricted group not scoped to the page, show restricted group in `Currently Viewing`',
+      isLoggedIn: false,
+      filtered_groups: [RESTRICTED_UNSCOPED_GROUP],
+    },
+    {
+      description:
+        'If logged out and only the Public group is present, show it in `Currently Viewing`',
+      isLoggedIn: false,
+      filtered_groups: [PUBLIC_GROUP],
+    },
+  ].forEach(({ description, isLoggedIn, filtered_groups }) => {
```

Comment:
Is there a case where the user is logged in and the group that is linked to is an unscoped restricted group of which the user is not a member meaning it would belong in the `Currently Viewing` section. (Not currently the case but I think this is where we potentially want to get to.)